### PR TITLE
Bitwise addition in C and C++ #1838 (GSSoC'20)

### LINKED ISSUE #1838: bitwise addition of two numbers in C and C++.(GSSoC'20)

### PR DESCRIPTION
**Pull request by DawoodBhai420**
Fixes #1838 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Iterative method to add two number by using bitwise operators.

#### Languages Used:
- C 
- C++

#### Files Added:
- bitwise_addition.c
- bitwise_addition.cpp

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by DawoodBhai420** [comment]
@singh-shreya6  can you please review the PR.

**Comment by DawoodBhai420** [comment]
@MastersAbh  can you please review this PR?

**Comment by raghav-dalmia** [comment]
Squash all the commits and create different issue for the recursive approach. Rest seems fine to me.

File: Bitwise Addition/bitwise_addition.c [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#include<stdio.h>
```
Comment (jainaman224) [inline Bitwise Addition/bitwise_addition.c]: Incorrect folder name.

**Comment by AJAYSHARMA2012** [comment]
I want to work on this issue as a participant of gssoc'20. Please assign this to me @jainaman224.

**Pull request merged**